crates/red_knot_python_semantic/src/types/subclass_of.rs

```diff
@@ -0,0 +1,62 @@
+use super::{ClassBase, ClassLiteralType, Db, KnownClass, Symbol, Type};
+
+/// A type that represents `type[C]`, i.e. the class literal `C` and class literals that are subclasses of `C`.
```

Comment:
This comment sort of implies a type as being made up of types, rather than objects:
```suggestion
/// A type that represents `type[C]`, i.e. the class object `C` and class objects that are subclasses of `C`.
```